assets/js/modules/analytics-4/components/audience-segmentation/dashboard/AudienceSelectionPanel/index.test.js

```diff
@@ -1729,11 +1752,9 @@ describe( 'AudienceSelectionPanel', () => {
 					measurementID: '56789',
 					webDataStreamID: '78901',
 				} );
+			} );
 
-				registry
-					.dispatch( MODULES_ANALYTICS_4 )
-					.receiveError( error, storeFunctionName, args );
-
+			afterEach( async () => {
```

Comment:
I think I understand what you're doing here, but this feels quite unorthodox compared to how we usually compose tests, namely:

1. We usually do not do assertions in `beforeEach` or `afterEach`.
2. All test cases usually read as natural language, e.g. "it should do something".

Could you possibly enlighten why we can't stick to the previous `it.each()` approach which was much cleaner?